apps/api/src/domain/models/digital-text/commands/translate-digital-text-title/translate-digital-text-title.command.integration.spec.ts

```diff
@@ -0,0 +1,180 @@
+    type: AggregateType.digitalText,
+    id: digitalTextId,
+};
+
+const originalLanguageCode = LanguageCode.English;
+
+const translationLanguageCode = LanguageCode.Chilcotin;
+
+const translationTitle = 'translation of title';
+
+const digitalTextCreated = new TestEventStream().andThen<DigitalTextCreated>({
+    type: DIGITAL_TEXT_CREATED,
+    payload: {
+        languageCodeForTitle: originalLanguageCode,
+    },
+});
+
+const digitalTextTitleTranslated = digitalTextCreated.andThen<DigitalTextTitleTranslated>({
+    type: 'DIGITAL_TEXT_TITLE_TRANSLATED',
+    payload: {
+        translation: translationTitle,
+        languageCode: translationLanguageCode,
+    },
+});
+
+const validPayload: TranslateDigitalTextTitle = {
+    aggregateCompositeIdentifier: digitalTextCompositeIdentifier,
+    languageCode: translationLanguageCode,
+    translation: translationTitle,
+};
+
+const validCommandFSA = {
+    type: commandType,
+    payload: validPayload,
+};
+
+const fsaFactory = new DummyCommandFsaFactory(() => validCommandFSA);
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app, databaseProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterAll(async () => {
+        await app.close();
+
+        databaseProvider.close();
+    });
+
+    describe(`when the command is valid`, () => {
+        it(`should succeed with the expected updates to the database`, async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA: () => validCommandFSA,
+                seedInitialState: async () => {
+                    const eventHistory = digitalTextCreated.as(digitalTextCompositeIdentifier);
+
+                    await app.get(ArangoEventRepository).appendEvents(eventHistory);
+                },
+                checkStateOnSuccess: async () => {
+                    const digitalTextSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.digitalText)
+                        .fetchById(digitalTextId);
+
+                    expect(digitalTextSearchResult).toBeInstanceOf(DigitalText);
```

Comment:
We also need to check that 
- the title has the new translation
- the event record has been persisted

Reply:
done